modules/blockexplorer/hashes.go

```diff
@@ -0,0 +1,224 @@
+package blockexplorer
+
+import (
+	"bytes"
+	"errors"
+
+	"github.com/NebulousLabs/Sia/crypto"
+	"github.com/NebulousLabs/Sia/encoding"
+	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/types"
+)
+
+const (
+	responseBlock        = "Block"
+	responseTransaction  = "Transaction"
+	responseFileContract = "FileContract"
+	responseOutput       = "Output"
+	responseAddress      = "Address"
+)
+
+// GetHashInfo returns sufficient data about the hash that was
+// provided to do more extensive lookups
+func (be *BlockExplorer) GetHashInfo(hash []byte) (interface{}, error) {
+	if len(hash) < crypto.HashSize {
+		return nil, errors.New("requested hash not long enough")
+	}
+
+	lockID := be.mu.RLock()
+	defer be.mu.RUnlock(lockID)
+
+	// Perform a lookup to tell which type of hash it is
+	typeBytes, err := be.db.GetFromBucket("Hashes", hash[:crypto.HashSize])
+	if err != nil {
+		return nil, err
+	}
+
+	var hashType int
+	err = encoding.Unmarshal(typeBytes, &hashType)
+	if err != nil {
+		return nil, err
+	}
+
+	switch hashType {
+	case hashBlock:
+		var id types.BlockID
+		copy(id[:], hash[:])
+		return be.db.getBlock(types.BlockID(id))
```

Comment:
if `len(id) == len(hash`), isn't a cast sufficient? i.e. `return be.db.getBlock(types.BlockID(hash)`?